### PR TITLE
Bugfix #3549/Fix generation of group index in WA

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- Fix generation of group index in WA

#### Please include detailed Test instructions for your pull request:
- WA should open

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/3974)

### CODE REVIEW

**Comment by mannycolon** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[src/js/actions/ProjectDataLoadingActions.js, line 36 at r1](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974#-L6s2eulDLbvgMCa_1K3:-L6s2eulDLbvgMCa_1K4:bgrgcbb) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/b66930908cb845cbb04d362a6ef12a9a7c3c01c6/src/js/actions/ProjectDataLoadingActions.js#L36)):*
> ```JavaScript
>         glDataDirectory = path.join(versionDirectory, 'kt');
>       } else {
>         glDataDirectory = ''; // don't try to load groups index for wordAlignment - it will ve generated
> ```

so we dont need groups index for word alignment tool?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974)*

**Comment by PhotoNomad0** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/js/actions/ProjectDataLoadingActions.js, line 36 at r1](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974#-L6s2eulDLbvgMCa_1K3:-L6s68vM1T4jmfw1XM9v:bovxziw) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/b66930908cb845cbb04d362a6ef12a9a7c3c01c6/src/js/actions/ProjectDataLoadingActions.js#L36)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

so we dont need groups index for word alignment tool?
</blockquote></details>

No - as stated in the comment, if it isn't found it gets generated.  that's done in getGroupsIndex()

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974)*

File: src/js/actions/ProjectDataLoadingActions.js [hunk 1]
```diff
@@ -28,8 +28,13 @@ export function loadProjectData(currentToolName) {
       const gatewayLanguage = projectDetailsReducer.currentProjectToolsSelectedGL[currentToolName]?projectDetailsReducer.currentProjectToolsSelectedGL[currentToolName]:'en';
       const dataDirectory = path.join(projectSaveLocation, '.apps', 'translationCore', 'index', currentToolName);
       const toolResourceDirectory = path.join(ospath.home(), 'translationCore', 'resources', gatewayLanguage, 'translationHelps', currentToolName);
-      const versionDirectory = ResourcesHelpers.getLatestVersionInPath(toolResourceDirectory);
-      const glDataDirectory = path.join(versionDirectory, 'kt');
+      let glDataDirectory = ''; // default is that if resource doesn't exist, groups index will be generated
+      if (currentToolName !== 'wordAlignment') { // wordAlignment groups index will be generated
```
Comment (richmahn) [inline src/js/actions/ProjectDataLoadingActions.js]: Can remove this if and move the inside lines out one indent.

File: src/js/actions/ProjectDataLoadingActions.js [hunk 2]
```diff
@@ -28,8 +28,13 @@ export function loadProjectData(currentToolName) {
...
+        const versionDirectory = ResourcesHelpers.getLatestVersionInPath(toolResourceDirectory);
+        if (versionDirectory) { // if resource version found, groups index will be generated
+          glDataDirectory = path.join(versionDirectory || '', 'kt');
```
Comment (richmahn) [inline src/js/actions/ProjectDataLoadingActions.js]: Shouldn't need the || if we know versionDirectory is not null.

**Comment by RoyalSix** [comment]
Reviewed 1 of 2 files at r2.
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/js/actions/ProjectDataLoadingActions.js, line 36 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974#-L6s2eulDLbvgMCa_1K3:-L6s9SoKcCZFlv0zYt9T:b-z6pemn) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/b66930908cb845cbb04d362a6ef12a9a7c3c01c6/src/js/actions/ProjectDataLoadingActions.js#L36)):*
<details><summary><i>Previously, PhotoNomad0 (Bruce McLean) wrote…</i></summary><blockquote>

No - as stated in the comment, if it isn't found it gets generated.  that's done in getGroupsIndex()
</blockquote></details>

True ^

---

*[src/js/actions/ProjectDataLoadingActions.js, line 35 at r2](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974#-L6s1-rbTpZmrqVmY5Zs-r2-35:-L6s9nfMd-MQokq-lA7z:b2ngthx) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/00f53b3c182a5c85fcb11b42fc86dac06095b4b2/src/js/actions/ProjectDataLoadingActions.js#L35)):*
<details><summary><i>Previously, richmahn (Richard Mahn) wrote…</i></summary><blockquote>

Shouldn't need the || if we know versionDirectory is not null.
</blockquote></details>

+1 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974)*

**Comment by RoyalSix** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/js/actions/ProjectDataLoadingActions.js, line 32 at r2](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974#-L6s1-rbTpZmrqVmY5Zs-r2-32:-L6s9qwZrBI3E3d-ZIj0:b-23hbn1) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/00f53b3c182a5c85fcb11b42fc86dac06095b4b2/src/js/actions/ProjectDataLoadingActions.js#L32)):*
<details><summary><i>Previously, richmahn (Richard Mahn) wrote…</i></summary><blockquote>

Can remove this if and move the inside lines out one indent.
</blockquote></details>

Should probably leave the if and just remove the `versionDirectory || '',`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974)*

**Comment by PhotoNomad0** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/js/actions/ProjectDataLoadingActions.js, line 32 at r2](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974#-L6s1-rbTpZmrqVmY5Zs-r2-32:-L6sBm56Ess9cVODyElT:b-hfpv0l) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/00f53b3c182a5c85fcb11b42fc86dac06095b4b2/src/js/actions/ProjectDataLoadingActions.js#L32)):*
<details><summary><i>Previously, RoyalSix (Sterling (Jay) Scott) wrote…</i></summary><blockquote>

Should probably leave the if and just remove the `versionDirectory || '',`
</blockquote></details>

removed

---

*[src/js/actions/ProjectDataLoadingActions.js, line 35 at r2](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974#-L6s1-rbTpZmrqVmY5Zs-r2-35:-L6sBoZn5RGBMWqAluro:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/00f53b3c182a5c85fcb11b42fc86dac06095b4b2/src/js/actions/ProjectDataLoadingActions.js#L35)):*
<details><summary><i>Previously, RoyalSix (Sterling (Jay) Scott) wrote…</i></summary><blockquote>

+1 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/unfoldingword-dev/translationcore/3974)*

**Comment by RoyalSix** [comment]
Code Review: `LGTM`
Local Testing:`LGTM`

**Comment by richmahn** [comment]
:lgtm:

---

Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/3974#-:-L6sDpge7d7U-aGbkY4M:bnfp4nl)*

**Pull request merged**